src/Text/CSL/Reference.hs

```diff
@@ -889,25 +889,51 @@ processCites rs cs
                         in res : procGr (a' ++ [[]]) xs
 
       procCs a [] = (a,[])
-      procCs a (c:xs)
-          | isIbid,  isLocSet = go "ibid-with-locator"
-          | isIbid            = go "ibid"
-          | isElem            = go "subsequent"
-          | otherwise         = go "first"
+      procCs a (c:xs) = let addCite    = init a ++ [last a ++ [c]]
```

Comment:
`init` and `last` are partial functions, too.  So this will break if `a` is null.

Reply:
Those aren't even mine! I'll give it a go though.

Reply:
Sorry, I didn't notice that at a quick glance. Well, there is a lot of ugly code in this code base, and cleaning it up as we work on it will help make it easier to figure out what is going on in the future.

Reply:
I think this whole function would be better off building the list from the front rather than from the back, don't you think?

Reply:
It worked out. Is there some kind of Haskell linter to run over the codebase that can catch these for us?